src/main/java/org/mitre/synthea/world/concepts/HealthRecord.java

```diff
@@ -454,6 +465,48 @@ public Encounter(long time, String type) {
       devices = new ArrayList<Device>();
       this.claim = new Claim(this, person);
     }
+
+    /**
+     * Add an observation to the encounter.
+     * @param time The time of the observation
+     * @param type The LOINC code for the observation
+     * @param value The observation value
+     */
+    public void addObservation(long time, String type, Object value) {
+      Observation observation = new Observation(time, type, value);
+      observation.codes.add(new Code("LOINC", type, type));
+      this.observations.add(observation);
+    }
```

Comment:
This seems duplicative of the `public Observation observation(long time, String type, Object value)` method. Even if you feel like it is significantly different (it only seems to have minor differences), it should be colocated with the existing method.

Reply:
They are different and I'm not sure that there is a good way to colocate these.

The root cause of the issue, IMHO, is the use of all of the inner classes in `HealthRecord`.

`addObservation` is for when you have an instance of a `HealthRecord.Encounter` and want to add an `HealthRecord.Observation` to that particular encounter. It is defined on `HealthRecord.Encounter`.

`observation` is defined on `HealthRecord`. You can only add to the current encounter. This prevents historical editing of the record.

I could pull the `observation` method up to just below the `HealthRecord.Encounter` class, but that breaks the pattern of having all of the `HealthRecord` instance methods at the bottom of the class.